wemake_python_styleguide/visitors/ast/keywords.py

```diff
@@ -121,6 +121,8 @@ def _iterate_returning_values(
         returns: List[ast.Return] = []
         has_values = False
         for sub_node in ast.walk(node):
+            if get_context(sub_node) != node:
```

Comment:
This code is now too complex, a bit of refactoring is required.

Reply:
yes, fixed